src/JNI/com/amazonaws/kinesis/video/producer/jni/Parameters.cpp

```diff
@@ -249,10 +251,194 @@ BOOL setClientInfo(JNIEnv *env, jobject clientInfo, PClientInfo pClientInfo) {
     return STATUS_FAILED(retStatus) ? FALSE : TRUE;
 }
 
+VOID setKvsRetryStrategy(JNIEnv *env, jobject kvsRetryStrategy, PKvsRetryStrategy pKvsRetryStrategy)
+{
+    jmethodID methodId = NULL;
+    jclass cls = NULL;
+    BOOL nullKvsRetryStrategy = FALSE;
+
+    CHECK(env != NULL && pKvsRetryStrategy != NULL);
+
+    if (kvsRetryStrategy == NULL) {
+        nullKvsRetryStrategy = TRUE;
+        goto CleanUp;
+    }
+
+    cls = env->GetObjectClass(kvsRetryStrategy);
+    if (cls == NULL) {
+        DLOGW("Failed to create Java kvsRetryStrategy class.");
+        nullKvsRetryStrategy = TRUE;
+        goto CleanUp;
+    }
+
+    methodId = env->GetMethodID(cls, "getRetryStrategy", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategy");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategy, setting pRetryStrategy to null.");
+        pKvsRetryStrategy->pRetryStrategy = NULL;
+    } else {
+        if (env->CallLongMethod(kvsRetryStrategy, methodId) == 0) {
+            pKvsRetryStrategy->pRetryStrategy = NULL;
+        } else {
+            // TODO: Implement once we have support for this in Java, setting to null for safety.
+            pKvsRetryStrategy->pRetryStrategy = NULL;
+        }
+    }
+    
+    methodId = env->GetMethodID(cls, "getRetryStrategyConfig", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategyConfig");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategyConfig, setting pRetryStrategyConfig to null.");
+        pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+    } else {
+        if (env->CallLongMethod(kvsRetryStrategy, methodId) == 0) {
+            pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+        } else {
+            // TODO: Implement once we have support for this in Java, setting to null for safety.
+            pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+        }
+    }
+
+    methodId = env->GetMethodID(cls, "getRetryStrategyType", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategyType");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategyType, setting retryStrategyType to EXPONENTIAL_BACKOFF_WAIT.");
+        pKvsRetryStrategy->retryStrategyType = KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT;
+    } else {
+        pKvsRetryStrategy->retryStrategyType = (KVS_RETRY_STRATEGY_TYPE) env->CallIntMethod(kvsRetryStrategy, methodId);
+    }
+
+
+CleanUp:
+    if (nullKvsRetryStrategy) {
+        DLOGI("Setting kvsRetryStrategy members to NULL/default values.");
+        pKvsRetryStrategy->pRetryStrategy = NULL;
+        pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+        pKvsRetryStrategy->retryStrategyType = KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT;
+    }
+}
+
+VOID setKvsRetryStrategyCallbacks(JNIEnv *env, jobject kvsRetryStrategyCallbacks, PKvsRetryStrategyCallbacks pKvsRetryStrategyCallbacks)
+{
+    jmethodID methodId = NULL;
+    jclass cls = NULL;
+    BOOL nullSetCallbacks = FALSE;
```

Comment:
Can we set this to TRUE instead and set to false wherever relevant? That will simplify this code block. Probably in the previous function too, but I will let you evaluate if that makes sense there.